packages/app/provider/index.tsx

```diff
@@ -1,13 +1,92 @@
+import AsyncStorage from '@react-native-async-storage/async-storage';
 import { config } from '@tloncorp/ui';
+import { useEffect, useState } from 'react';
+import React from 'react';
 import { TamaguiProvider, TamaguiProviderProps } from 'tamagui';
+import type { ThemeName } from 'tamagui';
+
+import { THEME_STORAGE_KEY } from '../constants';
+import { useIsDarkMode } from '../hooks/useIsDarkMode';
+
+export const ThemeContext = React.createContext<{
+  setActiveTheme: (theme: ThemeName) => void;
+}>({ setActiveTheme: () => {} });
 
 export function Provider({
   children,
   ...rest
 }: Omit<TamaguiProviderProps, 'config'>) {
+  const isDarkMode = useIsDarkMode();
+  const [activeTheme, setActiveTheme] = useState<ThemeName>(
+    isDarkMode ? 'dark' : 'light'
+  );
+
+  useEffect(() => {
+    const loadStoredTheme = async () => {
```

Comment:
I think boilerplate in this file could be reduced significantly using `keyValue`.